### ISSUE TITLE
OAUTH Bearer Token Refresh Issue?

### ISSUE THREAD

**Issue by Fishwaldo** [n8n-io/n8n#10199]
### Bug Description

Hi,
Not sure what the issue is here. We have a workflow connecting to CrowdStrike (I know!) and after about 1 hour after creating the credentials and the workflow being idle, I start getting "Invalid Bearer Token" errors back from the CrowdStrike API endpoint when its triggered. 

To fix the issue, I need to navigate to the CrowdStrike Credentials page and open it till I get the green "Connection tested successfully" message on the page (I don't need to modify anything, just open the page). 

The invalid Bearer Token Message appears after some idle time (about 1 hour) when either testing the workflow, or if the workflow is triggered (via schedule, or another API webhook). 

If I stop doing any testing, or there is no run of the workflow within 1 hour of "opening the crowdstrike credential page" then the invalid bearer token message appears. 

### To Reproduce

1. Create a Crowdstrike Node with Crowdstrike credentials
2. test - everything works
3. walk away for > 1 hour
4. test - "Invalid Bearer Token" from Crowdstrike node
5. go to credentials page - open crowdstrike, wait for green "connection tested successfully" message
6. test workflow - everything works
7. goto step 3 :)

### Expected behavior

Seems the credentials need to be "refreshed" before a node is executed if its outside of some window (1 hour?)

### Operating System

Cloud

### n8n Version

1.51.1

### Node.js Version

?

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @Fishwaldo,

Oddly enough I was sure I fixed this about 3 months ago, Out of interest how are you using the http request node is it with the ignore errors option?

**Comment by Fishwaldo** [comment]
Nope... its stop on error - But the "invalid bearer token" comes as a normal output, not a error:
![image](https://github.com/user-attachments/assets/8493e2d2-e2a4-4ce1-9448-a1d78994a219)

**Comment by Fishwaldo** [comment]
Btw, I’m using the Crowdstrike node, but I guess it’s just the http request node underneath?

**Comment by Joffcom** [comment]
Hey @Fishwaldo,

Yeah it uses the HTTP Request node under it, Can you share the node JSON? Looking at the credenital we have not changed it so it should be working so I am wondering if there is a setting or something that is not triggering the refresh. One of the problematic things with Crowdstrike is trying to get an account for testing from them so this may take a bit of time to resolve.

**Comment by Fishwaldo** [comment]
Hi @Joffcom 
Here is the first node that executes a call to CrowdStrike:

```{
  "meta": {
    "instanceId": "10920e7b74683a53d64cdfb0acfb43615354632e02140aff66b59390a4bf18a9"
  },
  "nodes": [
    {
      "parameters": {
        "curlImport": "",
        "httpVariantWarning": "",
        "method": "GET",
        "url": "https://api.us-2.crowdstrike.com/<redacted>,
        "": "",
        "authentication": "predefinedCredentialType",
        "nodeCredentialType": "crowdStrikeOAuth2Api",
        "provideSslCertificates": false,
        "sendQuery": true,
        "specifyQuery": "keypair",
        "queryParameters": {
          "parameters": [
            {
              "name": "filter",
              "value": "rule.topic:’<redacted>’”
            }
          ]
        },
        "sendHeaders": false,
        "sendBody": false,
        "options": {
          "pagination": {
            "pagination": {
              "paginationMode": "updateAParameterInEachRequest",
              "webhookNotice": "",
              "parameters": {
                "parameters": [
                  {
                    "type": "qs",
                    "name": "offset",
                    "value": "={{ $response.body.meta.pagination.offset + 100 }}"
                  }
                ]
              },
              "paginationCompleteWhen": "other",
              "completeExpression": "={{ $response.body.meta.pagination.offset + $response.body.meta.pagination.limit >= $response.body.meta.pagination.total }}",
              "limitPagesFetched": false,
              "requestInterval": 0
            }
          }
        },
        "infoMessage": ""
      },
      "id": "3aeadee2-f935-49f4-acc2-65f5900b32e9",
      "name": "Get Notifications",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 4.2,
      "position": [
        -720,
        -240
      ],
      "extendsCredential": "crowdStrikeOAuth2Api",
      "credentials": {
        "crowdStrikeOAuth2Api": {
          "id": "m3KLJ1lHlffkXW2z",
          "name": "CrowdStrike account"
        }
      }
    }
  ],
  "connections": {},
  "pinData": {}
}```

**Comment by Fishwaldo** [comment]
If you can drop me a mail (I sent one for #10181) I might be able to hook you up with someone at CrowdStrike to get you a account for testing.

**Comment by alonso-chainels** [comment]
Hey @Joffcom - actually I think something is happening with the HTTP Request node as it is not auth code with the refresh tokens. I am using OAuth2 and in my case all the workflows I had running with at least 1 month are breaking. The solution is to open the credentials and hit the **"Reconnect button"**. I cannot share information here but of course we can review it together if necessary by email or even a meeting.

**Comment by Joffcom** [comment]
Hey @alonso-chainels,

That is interesting, Which credentials are you using is it a predefined service or custom oauth?

**Comment by Joffcom** [comment]
@alonso-chainels out of interest are you using the pagination option in an http request node?

**Comment by mredodos** [comment]
at my end have same issue with oauth 2 x.com and click on reconnect not work and not solve the problem.

**Comment by EthanHemo** [comment]
I also encounter this issue. 
I created a "Generic Credential Type" > "OAuth2 API", chose "Client Credentials", set details, ran the API and it worked.
After couple of hours tried to use the same credential, and got failed credentials with "perhaps check your credentials?"
I recreated the credentials and it worked again. 
I suspect the refresh isn't happen but fetch an old token.

Is there a way to prioritize this issue? it seems a bit old :(

**Comment by Joffcom** [comment]
Hey @Fishwaldo,

Thank you for reaching out! We’ve received your issue and are looking into it. To help us track this internally, we’ve created a Linear ticket with the reference: "GHC-3690".

We’ll keep you updated as we make progress, but if you have any additional details or context to share, feel free to add them here - it’s always helpful!

Thanks for your patience and for bringing this to our attention.
